CMakeLists.txt

```diff
@@ -467,10 +467,12 @@ if(SWIFT_PATH_TO_CMARK_BUILD)
 endif()
 message(STATUS "")
 
-if("${SWIFT_NATIVE_LLVM_TOOLS_PATH}" STREQUAL "")
-  set(SWIFT_CROSS_COMPILING FALSE)
```

Comment:
@shahmishal, I believe the only reason this was set was because `SWIFT_NATIVE_LLVM_TOOLS_PATH` was used to find clang in `stdlib/CMakeLists.txt` below. However, since that was a mistake that I tried to fix in #34437, I now use this clang path check and `SWIFT_PREBUILT_CLANG` here and in `cmake/modules/SwiftSharedCMakeConfig.cmake` instead. I believe this `SWIFT_PREBUILT_CLANG` is a better variable name as the clang might not just be passed in for cross-compiling, but also for natively building the standalone stdlib, as this pull was originally meant for.